controllers/openstackdeploy_controller.go

```diff
@@ -0,0 +1,235 @@
+type OpenStackDeployReconciler struct {
+	client.Client
+	Kclient kubernetes.Interface
+	Log     logr.Logger
+	Scheme  *runtime.Scheme
+}
+
+// GetClient -
+func (r *OpenStackDeployReconciler) GetClient() client.Client {
+	return r.Client
+}
+
+// GetKClient -
+func (r *OpenStackDeployReconciler) GetKClient() kubernetes.Interface {
+	return r.Kclient
+}
+
+// GetLogger -
+func (r *OpenStackDeployReconciler) GetLogger() logr.Logger {
+	return r.Log
+}
+
+// GetScheme -
+func (r *OpenStackDeployReconciler) GetScheme() *runtime.Scheme {
+	return r.Scheme
+}
+
+// +kubebuilder:rbac:groups=osp-director.openstack.org,resources=openstackdeploys,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=osp-director.openstack.org,resources=openstackdeploys/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=osp-director.openstack.org,resources=openstackdeploys/finalizers,verbs=update
+// +kubebuilder:rbac:groups=batch,resources=jobs,verbs=get;list;create;update;delete;watch
+
+// Reconcile - OpenStackDeploy
+func (r *OpenStackDeployReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	_ = log.FromContext(ctx)
+
+	// Fetch the ProvisionServer instance
+	instance := &ospdirectorv1beta1.OpenStackDeploy{}
+	err := r.Client.Get(context.TODO(), req.NamespacedName, instance)
+	if err != nil {
+		if k8s_errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected.
+			// For additional cleanup logic use finalizers. Return and don't requeue.
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return ctrl.Result{}, err
+	}
+
+	//
+	// initialize condition
+	//
+	cond := &ospdirectorv1beta1.Condition{}
+
+	//
+	// Used in comparisons below to determine whether a status update is actually needed
+	//
+	currentStatus := instance.Status.DeepCopy()
+	statusChanged := func() bool {
+		return !equality.Semantic.DeepEqual(
+			r.getNormalizedStatus(&instance.Status),
+			r.getNormalizedStatus(currentStatus),
+		)
+	}
+
+	defer func(cond *ospdirectorv1beta1.Condition) {
+		//
+		// Update object conditions
+		//
+		instance.Status.CurrentState = ospdirectorv1beta1.DeployState(cond.Type)
+		instance.Status.CurrentReason = ospdirectorv1beta1.DeployReason(cond.Message)
+
+		instance.Status.Conditions.UpdateCurrentCondition(
+			cond.Type,
+			cond.Reason,
+			cond.Message,
+		)
+
+		if statusChanged() {
+			if updateErr := r.Client.Status().Update(context.Background(), instance); updateErr != nil {
+				if err == nil {
+					err = common.WrapErrorForObject(
+						"Update Status", instance, updateErr)
+				} else {
+					common.LogErrorForObject(r, updateErr, "Update status", instance)
+				}
+			} else {
+				// log status changed messages also to operator log
+				common.LogForObject(r, cond.Message, instance)
+			}
```

Comment:
for the other controllers we switched to always log the condition to the operator log, even if it has not changed so that its better to follow the reconciliations from the operator log [1]

[1] https://github.com/openstack-k8s-operators/osp-director-operator/blob/master/controllers/openstacknetconfig_controller.go#L128-L139